editoast/src/views/infra/mod.rs

```diff
@@ -412,12 +437,47 @@ async fn unlock(infra: Path<i64>, db_pool: Data<DbPool>) -> Result<HttpResponse>
     Ok(HttpResponse::NoContent().finish())
 }
 
+#[derive(Debug, Default, Deserialize)]
+pub struct Payload {
+    infra: i64,
+}
+
+/// Builds a Core infra load request, runs it
+async fn call_core_infra_load(
+    payload: Json<Payload>,
+    db_pool: Data<DbPool>,
+    core: Data<CoreClient>,
+) -> Result<()> {
+    let payload = payload.into_inner();
+    let infra_id = payload.infra;
+    let infra = Infra::retrieve(db_pool.clone(), infra_id)
+        .await?
+        .ok_or(InfraApiError::NotFound { infra_id })?;
+    let infra_request = InfraLoadRequest {
+        infra: infra.id.unwrap(),
+        expected_version: infra.version,
+    };
+    infra_request.fetch(core.as_ref()).await?;
+    Ok(())
+}
+
+#[post("/infra_load")]
+async fn infra_load(
+    payload: Json<Payload>,
+    db_pool: Data<DbPool>,
+    core: Data<CoreClient>,
+) -> Result<String> {
+    call_core_infra_load(payload, db_pool, core).await?;
+    Ok("infra loaded".into())
```

Comment:
It's just my opinion but I feel like if the endpoint doesn't have anything meaningful to return, then it should return nothing. `StatusCode::NO_CONTENT` seems more appropriate to me here.

Reply:
Yes you're write !